src/firmware_new/boards/frankie_v1/io/proto_multicast_communication.c

```diff
@@ -0,0 +1,230 @@
+        (ProtoMulticastCommunicationProfile_t*)malloc(
+            sizeof(ProtoMulticastCommunicationProfile_t));
+
+    const osMutexAttr_t mutex_attr = {profile_name, osMutexPrioInherit, NULL, 0U};
+
+    profile->profile_name        = profile_name;
+    profile->port                = port;
+    profile->message_fields      = message_fields;
+    profile->message_max_size    = message_max_size;
+    profile->protobuf_struct     = protobuf_struct;
+    profile->profile_mutex       = osMutexNew(&mutex_attr);
+    profile->communication_event = osEventFlagsNew(NULL);
+    ip6addr_aton(multicast_address, &profile->multicast_address);
+
+    return profile;
+}
+
+void io_proto_multicast_sender_Task(void* arg)
+{
+    ProtoMulticastCommunicationProfile_t* comm_profile =
+        (ProtoMulticastCommunicationProfile_t*)arg;
+
+    osEventFlagsWait(networking_event, NETIF_CONFIGURED, osFlagsWaitAny, osWaitForever);
+
+    // Bind the socket to the multicast address and port we then use that
+    // communication profile to join the specified multicast group.
+    //
+    // We use IP6_ADDR_ANY which will default to the ethernet interface on the STM32H7
+    struct netconn* conn = netconn_new(NETCONN_UDP_IPV6);
+    netconn_bind(conn, IP6_ADDR_ANY, comm_profile->port);
+    netconn_join_leave_group(conn, &comm_profile->multicast_address, IP6_ADDR_ANY,
+                             NETCONN_JOIN);
+
+    // this buffer is used to hold serialized proto
+    uint8_t buffer[comm_profile->message_max_size];
+
+    // network buffer
+    struct netbuf* tx_buf = netbuf_new();
+
+    for (;;)
+    {
+        pb_ostream_t stream = pb_ostream_from_buffer(buffer, sizeof(buffer));
+
+        // block until protobuf has been updated
+        io_proto_multicast_communication_profile_blockUntilEvents(comm_profile,
+                                                                  PROTO_UPDATED);
+
+        // serialize proto into buffer
+        io_proto_multicast_communication_profile_acquireLock(comm_profile);
+        pb_encode(&stream, comm_profile->message_fields, comm_profile->protobuf_struct);
+        io_proto_multicast_communication_profile_releaseLock(comm_profile);
+
+        netbuf_alloc(tx_buf, stream.bytes_written);
+
+        // package payload and send over udp
+        tx_buf->p->payload = buffer;
+        netconn_sendto(conn, tx_buf, &comm_profile->multicast_address,
+                       comm_profile->port);
+    }
+
+    netbuf_delete(tx_buf);
+}
+
+void io_proto_multicast_listener_Task(void* arg)
+{
+    ProtoMulticastCommunicationProfile_t* comm_profile =
+        (ProtoMulticastCommunicationProfile_t*)arg;
+
+    osEventFlagsWait(networking_event, NETIF_CONFIGURED, osFlagsWaitAny, osWaitForever);
+
+    // Bind the socket to the multicast address and port we then use that comm_profile
+    // to join the specified multicast group.
+    struct netconn* conn = netconn_new(NETCONN_UDP_IPV6);
+
+    netconn_set_ipv6only(conn, true);
```

Comment:
do we not need to specify these settings on the send task as well?

Reply:
yup we do, good catch fixed

its more of a safety thing, didn't seem to affect functionality